tests/test_predictor.py

```diff
@@ -44,3 +46,31 @@ def test_boston_dataset(max_bins):
 
     assert r2_score(y_train, predictor.predict(X_train)) > 0.85
     assert r2_score(y_test, predictor.predict(X_test)) > 0.70
+
+
+def test_pre_binned_data():
+    # Make sure ValueError is raised when predictor.predict() is called while
+    # the predictor does not have any numerical thresholds.
+
+    X, y = make_regression()
+
+    # Init gradients and hessians to that of least squares loss
+    gradients = -y.astype(np.float32)
+    hessians = np.ones(1, dtype=np.float32)
+
+    mapper = BinMapper(random_state=0)
+    X_binned = mapper.fit_transform(X)
+    grower = TreeGrower(X_binned, gradients, hessians,
+                        n_bins_per_feature=mapper.n_bins_per_feature_)
+    grower.grow()
+    predictor = grower.make_predictor(
+        numerical_thresholds=None
+    )
+
+    assert_raises_regex(
+        ValueError,
+        'This predictor does not have numerical thresholds',
+        predictor.predict, X_binned
+    )
+
+    predictor.predict_binned(X)  # No error
```

Comment:
Yes you're right, I've added the check.

And yes all the checks in `PredictorNode` are redundant since `GradientBoosting.predict()` would only call `predict_binned()` on `uint8` data and `predict()` on `non-uint8` data.

My guess is that the overhead is negligible compared to the prediction time, but no strong opinion, we can remove them and just add a comment.